provider/aws: use single import path for aws-sdk-go

rjeczalik opened the pull request:
Even though github.com/awslabs is an alias for github.com/aws and both paths reference the same package, mixing both duplicates aws-sdk-go packages that need to be vendored (e.g. [this](https://github.com/hashicorp/terraform/blob/master/deps/v0-6-6.json#L207) and [that](https://github.com/hashicorp/terraform/blob/master/deps/v0-6-6.json#L42)).

This CL fixes the above issue and may by accident fix the coverage script, however I did not test it.

rjeczalik commented:
I've also fixed your `updatedeps`, since it was overwriting the build directory with `go get`'d terraform from master - see build logs: https://travis-ci.org/hashicorp/terraform/jobs/89503522#L441

rjeczalik commented:
@jen20 @catsby Is there a chance this PR gets merged and included in 0.6.7 release? Not pushing, just asking, not sure what's left for 0.6.7.

radeksimko commented:
I think changing the imports make total sense to me and I'd merge _that commit_ straight away. I don't quite know the reasons behind the _fairly complicated_ dependency update logic, but I think there was a few reasons.

I don't think the errors in the build log you are referring to are consistent - I relaunched the build a while ago and it seems to be ok.

rjeczalik commented:
> I don't think the errors in the build log you are referring to are consistent - I relaunched the build a while ago and it seems to be ok.

@radeksimko Got it, I've rebased with master and removed the second commit.

radeksimko commented:
ok, this is really strange, the tests were passing before (_without_ the commit modifying the `Makefile`).
This almost looks like a Travis bug... all the tests pass on my machine, I think we'll have to isolate this and figure out what's happening there.

rjeczalik commented:
@radeksimko See the above build log - the `updatedeps` does overwrite build directory with `go get`'d project from master.

Now a word of explanation why the failed build went green when you clicked "rebuild" - looking at how travis checks the source code for the build:

```
git clone --depth=50 https://github.com/hashicorp/terraform.git hashicorp/terraform
git fetch origin +refs/pull/3773/merge
git checkout -qf FETCH_HEAD
```

It will always pull the HEAD of the PR, so in fact the failed build was restarted with my fix, that's why it went green.

rjeczalik commented:
@radeksimko Please update me how we proceed, I'd remove the debug commit and send my fix  for `updatedeps`.

radeksimko commented:
Thanks for the debugging & explanation & repro steps, I will try to reproduce this locally.

It looks like this line is being ignored for some reason... https://github.com/hashicorp/terraform/blob/master/Makefile#L53

If we're changing the `updatedeps` target, we should review the original reasons behind the current state, e.g. https://github.com/hashicorp/terraform/commit/a764adbf1b24a6a61b783a2aa8b8ac00f2b8cabd

rjeczalik commented:
@radeksimko I **guess** the reason behind such complicated logic was missing packages while building tests - `go get` won't import test-only packages without the `-t` flag and I think the logic tried to workaround that.

`go get -t -f -u -v ./...` will work exactly the same as the long bash command.

rjeczalik commented:
The failure at the tip is caused by https://github.com/golang/go/issues/13206.

radeksimko commented:
I believe the reason behind overwriting the build directory is https://github.com/hashicorp/go-getter which has dependency on terraform https://github.com/hashicorp/go-getter/search?utf8=%E2%9C%93&q=terraform -> that creates a dependency circle.

The `go-getter` library was created about 1 month ago which explains why we never experienced troubles with the current `updatedeps` - there was never a circular dependency.

rjeczalik commented:
@radeksimko Rebased with master, #3922 fixes the problem with overwriting the build dir.

radeksimko commented:
LGTM. :+1:
